staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -72,6 +74,10 @@ type AvailableConditionController struct {
 	syncFn func(key string) error
 
 	queue workqueue.RateLimitingInterface
+	// map from service-namespace -> service-name -> apiservice names
+	cache map[string]map[string][]string
```

Comment:
an alternative will be inserting another indexer into the apiService informer e.g. https://github.com/kubernetes/kubernetes/blob/7e806a2dd6efb156858706aca78f8a9850315f7b/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L368-L380

so that we dont need to deal w/ the cache managed in a nested-map in addition to the lock instance. an indexer and index function will achieve achieve the same @liggitt wdyt?